src/main/java/com/zendesk/maxwell/producer/MaxwellKinesisProducer.java

```diff
@@ -51,7 +53,11 @@ public void onFailure(Throwable t) {
 
 		logger.error("Exception during put", t);
 
-		cc.markCompleted();
+		if (!context.getConfig().ignoreProducerError) {
+			context.terminate(new RuntimeException(t));
```

Comment:
Ugh, true. This'll do then - not like anyone's going to be catching it anyway 🤷‍♂️